Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/overview/items/PersonCertificateCard.kt

```diff
@@ -25,48 +25,88 @@ class PersonCertificateCard(parent: ViewGroup) :
         PersonOverviewItemBinding.inflate(layoutInflater, itemView.findViewById(R.id.card_container), true)
     }
 
-    // TODO: Update curItem with the right certificate
     override val onBindData: PersonOverviewItemBinding.(
         item: Item,
         payloads: List<Any>
     ) -> Unit = { item, payloads ->
         val curItem = payloads.filterIsInstance<Item>().singleOrNull() ?: item
 
-        val color = when {
-            curItem.admissionState.primaryCertificate!!.isValid -> curItem.colorShade
-            else -> PersonColorShade.COLOR_INVALID
-        }
-        name.text = curItem.admissionState.primaryCertificate.fullName
+        when (curItem.admissionState) {
+            is TwoG -> {
+                setUIState(
+                    primaryCertificate = curItem.admissionState.twoGCertificate,
+                    colorShade = curItem.colorShade,
+                    isCombinedCertificate = false,
+                    statusBadgeText = R.string.confirmed_status_2g_badge,
+                    badgeCount = curItem.badgeCount,
+                    onCovPassInfoAction = curItem.onCovPassInfoAction
+                )
+            }
+            is TwoGPlusPCR -> {
+                setUIState(
+                    primaryCertificate = curItem.admissionState.twoGCertificate,
+                    secondaryCertificate = curItem.admissionState.testCertificate,
+                    colorShade = curItem.colorShade,
+                    isCombinedCertificate = true,
+                    statusBadgeText = R.string.confirmed_status_2g_plus_badge,
+                    badgeCount = curItem.badgeCount,
+                    onCovPassInfoAction = curItem.onCovPassInfoAction
+                )
+            }
+
+            is TwoGPlusRAT -> {
```

Comment:
This calls the same arguments as the block above. We can do 
```kotlin
is TwoGPlusPCR, is TwoGPlusRAT -> { ...}
```

Reply:
I tried that and but then I get Unresolved reference for twoGCertificate & testCertificate.

Reply:
Ah I see, the smart cast isn't working in this case anymore. Then we should leave it as is.